File: backend-cs/Models/Comment.cs
namespace SocialFeed.Api.Models;

public class Comment
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Text { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string PostId { get; set; } = string.Empty;
    public string AuthorId { get; set; } = string.Empty;

    public Post Post { get; set; } = null!;
    public User Author { get; set; } = null!;
}

File: android/app/src/main/java/com/example/socialfeed/data/db/entity/Comment.kt
package com.example.socialfeed.data.db.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "comments",
    foreignKeys = [
        ForeignKey(entity = Post::class, parentColumns = ["id"], childColumns = ["postId"], onDelete = ForeignKey.CASCADE),
        ForeignKey(entity = User::class, parentColumns = ["id"], childColumns = ["authorId"], onDelete = ForeignKey.CASCADE)
    ],
    indices = [Index("postId"), Index("authorId")]
)
data class Comment(
    @PrimaryKey val id: String,
    val postId: String,
    val authorId: String,
    val text: String,
    val createdAt: Long = System.currentTimeMillis()
)
